hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncNonMetaRegionLocator.java

```diff
@@ -450,13 +601,11 @@ private RegionLocations locateRowInCache(TableCache tableCache, TableName tableN
   private RegionLocations locateRowBeforeInCache(TableCache tableCache, TableName tableName,
     byte[] row, int replicaId) {
     boolean isEmptyStopRow = isEmptyStopRow(row);
-    Map.Entry<byte[], RegionLocations> entry =
-      isEmptyStopRow ? tableCache.cache.lastEntry() : tableCache.cache.lowerEntry(row);
-    if (entry == null) {
+    RegionLocations locs = tableCache.regionLocationCache.findForBeforeRow(row);
```

Comment:
Do we still need the below check for whether the row is contained in the region? I suppose this should be done in findForBeforeRow already?

Reply:
Moved this logic into findForBeforeRow (and similar for findForRow)